qtism/data/storage/xml/marshalling/ObjectMarshaller.php

```diff
@@ -52,12 +52,22 @@ protected function unmarshallChildrenKnown(DOMElement $element, QtiComponentColl
             $component = new $fqClass($data, $type);
             $component->setContent(new ObjectFlowCollection($children->getArrayCopy()));
 
-            if (($width = $this->getDOMElementAttributeAs($element, 'width', 'integer')) !== null) {
-                $component->setWidth($width);
+            if (($height = self::getDOMElementAttributeAs($element, 'height', 'string')) !== null) {
```

Comment:
Bonus points: you can reduce cognitive load by separating assignment from conditional:
```
$height = self::getDOMElementAttributeAs($element, 'height', 'string');
if ($height !== null) {
```

Reply:
Done